openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -202,7 +202,21 @@ def get_publishable_entity_by_key(learning_package_id, /, key) -> PublishableEnt
     )
 
 
-def get_last_publish(learning_package_id: int, /) -> PublishLog | None:
+def get_last_publish(learning_package_id: int, /, key=None) -> PublishLog | None:
+    """
+    Return the most recent PublishLog for the given LearningPackage + optional PublishableEntity.
+
+    Returns None if no related PublishLog (or PublishLogRecord) is found.
+    """
+    if key:
+        publish_log_record = PublishLogRecord.objects \
+                                             .filter(publish_log__learning_package_id=learning_package_id) \
+                                             .filter(entity__key=key) \
+                                             .order_by('-publish_log__id') \
+                                             .select_related('publish_log') \
+                                             .first()
+        return publish_log_record.publish_log if publish_log_record else None
```

Comment:
Could you please implement this as a new [`VersioningHelper`](https://github.com/openedx/openedx-learning/blob/main/openedx_learning/apps/authoring/publishing/model_mixins.py#L66) method (maybe `last_published_at`?) and follow the relationships:

`Component.publishable_entity -> PublishableEntity.published -> Published.publish_log_record -> PublishLogRecord.publish_log -> PublishLog. published_at`

Reply:
1. Sounds good.
2. Please put it in `PublishableentityMixinManager` for now. Thank you.

Reply:
@ormsbee Hmm.. modifying `PublishableEntityMixinManager` isn't working, because the components are fetched using [`with_publishing_relations`](https://github.com/openedx/openedx-learning/blob/01defb1cc871b890df46e3ead46c9b7382155131/openedx_learning/apps/authoring/components/models.py#L130) instead of with the base `PublishableEntityMixinManager`. Should I add it to `with_publishing_relations` instead?

Reply:
Ah, yeah, `with_publishing_relations` makes more sense anyhow. Thank you.